src/contracts/collections/pool_keys.rs

```diff
@@ -34,38 +30,25 @@ impl PoolKeys {
         Ok(())
     }
 
-    #[allow(dead_code)]
-    pub fn remove(&mut self, pool_key: PoolKey) -> Result<(), InvariantError> {
-        match self.get_index(pool_key) {
-            Some(index) => {
-                self.pool_keys_by_index.remove(index);
-                self.pool_keys_length =
-                    self.pool_keys_length
-                        .checked_sub(1)
-                        .ok_or(InvariantError::SubUnderflow(
-                            self.pool_keys_length as u128,
-                            1,
-                        ))?;
-                self.pool_keys.remove(pool_key);
-                Ok(())
-            }
-            None => Err(InvariantError::PoolKeyNotFound),
-        }
-    }
-
     pub fn contains(&self, pool_key: PoolKey) -> bool {
         self.pool_keys.get(pool_key).is_some()
     }
 
     pub fn get_all(&self, size: u16, offset: u16) -> Vec<PoolKey> {
         let offset_with_size = offset.checked_add(size).unwrap();
 
-        let max = if offset_with_size > self.pool_keys_length {
+        let upper_bound = if offset_with_size > self.pool_keys_length {
             self.pool_keys_length
         } else {
             offset_with_size
         };
 
+        let max = if upper_bound.checked_sub(offset).unwrap() > MAX_POOL_KEYS_RETURNED {
```

Comment:
same thing with `get_all` function in positions collection would be helpful

Reply:
Yes, we should add this to the `get_all` function in the `positions` collection as well